riscv: telink:tl3218x: add ble optimize.

haiwentelink opened the pull request:
- add optimize for ble.
- low down the ramcost.

s07641069 commented on config/telink/chip-module/Kconfig.defaults:
```diff
@@ -34,10 +34,12 @@ choice LOG_MODE
 endchoice
 
 config CHIP_APP_LOG_LEVEL
+    default 4 if SOC_RISCV_TELINK_TL321X # certify should be 2 
```
Clarify this and change or revert
Currently both lines has default values 4

haiwentelink replied:
this is just a comment , liang find the value relate to the certify , so it's just a mark to notice us . 
If you think there is no need to do this , i can delete .

s07641069 replied:
remove, we don't need duplicated default values 
just add when it will be different

s07641069 commented on config/telink/chip-module/Kconfig.defaults:
```diff
@@ -113,7 +115,7 @@ config GPIO
 
 config BT
-    default n if SOC_SERIES_RISCV_TELINK_TLX
+    default y if SOC_SERIES_RISCV_TELINK_TLX
```
don't need this, default value already = y
remove this
is was set to n just temporary

haiwentelink replied:
OK, I will do this .

s07641069 commented on config/telink/chip-module/Kconfig.defaults:
```diff
@@ -175,7 +177,7 @@ config BT_DEVICE_NAME_GATT_WRITABLE
     default n
 
-if SOC_SERIES_RISCV_TELINK_B9X
+if SOC_SERIES_RISCV_TELINK_B9X || SOC_SERIES_RISCV_TELINK_TLX
```
Looks like you are using B9X_BLE_* under TLX BLE stack
Below I prepared clean code for TLX BLE, but in case if you are using same B9X_BLE_* for now then just remove my code:

if SOC_SERIES_RISCV_TELINK_TLX
...............................
endif # TLX_BLE_CTRL_MASTER_MAX_NUM

haiwentelink replied:
agree , already notice this problem ,just wait for response .

haiwentelink committed:
riscv: telink:tl3218x: add ble optimize.

- add optimize for ble.
- low down the ramcost.
- revert nvs cache to pass certify.

Signed-off-by: Haiwen Xia <haiwen.xia@telink-semi.com>

haiwentelink commented:
can not set the cache size to 8k , otherwise the tl3218x compile will fail for ram not enough . the lighting-app will open the ota, factorydata and shell , without shell is enough , but with shell will exceed 1312bytes .
